### PR TITLE
PMAC device to ophyd_async 

### PR DESCRIPTION
**Pull request by noemifrisina**
Needed for [Mx#98](https://github.com/DiamondLightSource/mx_bluesky/issues/98)

Convert PMAC device to ophyd_async and add functionality to set various PMAC_strings.

### Instructions to reviewer on how to test:
1. Tests pass

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly

### CODE REVIEW

File: src/dodal/devices/i24/pmac.py [hunk 1]
```diff
@@ -1,19 +1,78 @@
...
+    set up.
+    For example, for laser1:
+        Use M712 = 0 if triggering on falling edge.
+        Use M712 = 1 if on rising edge.
+    """
+
+    LASER1ON = " M712=1 M711=1"
+    LASER1OFF = " M712=0 M711=1"
+    LASER2ON = " M812=1 M811=1"
+    LASER2OFF = " M812=0 M811=1"
```
Comment (DominicOram) [inline src/dodal/devices/i24/pmac.py:28]: Nit: I think some `_` would make these more readable:
```suggestion
    LASER_1_ON = " M712=1 M711=1"
    LASER_1_OFF = " M712=0 M711=1"
    LASER_2_ON = " M812=1 M811=1"
    LASER_2_OFF = " M812=0 M811=1"
```

File: src/dodal/devices/i24/pmac.py [hunk 2]
```diff
@@ -1,19 +1,78 @@
...
+
+
+class PMACStringMove(Triggerable):
```
Comment (DominicOram) [inline src/dodal/devices/i24/pmac.py:31]: Could: A docstring on this would be nice

**Review by DominicOram — CHANGES_REQUESTED** [review]
Great, thank you. Some comments in code

**Review by dperl-dls — APPROVED** [review]
lgtm, thanks

**Pull request merged**